File: 73-set-matrix-zeroes/set-matrix-zeroes.cpp
class Solution {
public:
    void setZeroes(vector<vector<int>>& matrix) {
        queue<pair<int, int>> q;
        int m = matrix.size();
        int n = matrix[0].size();

        for (int i = 0; i < m; i++) {
            for (int j = 0; j < n; j++) {
                if (matrix[i][j] == 0) {
                    q.push({i,j});
                }
            }
        }
        while(!q.empty())
        {
            int x=q.front().first;
            int y=q.front().second;
            q.pop();

            int ind = x-1;
            while(ind>=0)
            {
                matrix[ind][y]=0;
                ind--;
            }
            ind= x+1;
            while(ind<=m-1)
            {
                matrix[ind][y]=0;
                ind++;
            }
            ind =y-1;
            while(ind>=0)
            {
                matrix[x][ind]=0;
                ind--;
            }
            ind= y+1;
            while(ind<=n-1)
            {
                matrix[x][ind]=0;
                ind++;
            }
        }
    }
};

File: 73-set-matrix-zeroes/set-matrix-zeroes.java
class Solution {
    public class Pair{
        int x;
        int y;
        Pair(int x, int y)
        {
            this.x= x;
            this.y=y;
        }

    }
    public void setZeroes(int[][] matrix) {
        Queue<Pair>q =new LinkedList<>();
        int n=matrix.length;
        int m=matrix[0].length;
        for(int i=0;i<n;i++)
        {
            for(int j=0;j<m;j++)
            {
                if(matrix[i][j]==0)
                {
                    q.add(new Pair(i,j));
                }
            }
        }
        while(!q.isEmpty())
        {
           Pair p = q.poll();
           System.out.println(p.x+ " "+p.y);
            int x=p.x;
            int y= p.y;
        
            int ind= x-1;
            while(ind>=0)
            {
                matrix[ind][y]=0;
                ind--;
            }
            ind=x+1;
            while(ind<n)
            {
                matrix[ind][y]=0;
                ind++;
            }
            ind =y-1;
            while(ind>=0)
            {
                matrix[x][ind]=0;
                ind--;
            }
            ind=y+1;
            while(ind<m)
            {
                matrix[x][ind]=0;
                ind++;
            }
        }
        
        
    }
}